wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow]  Update Welcome Screen design

Fixes Partially #17368

- This PR implements the latest design updates to the Welcome Screen of the Jetpack migration flow,
- It also implements the processing state which is visible after tapping the `Continue` button,
- The screen orientation was locked to portrait for all migration flow steps.

Refactoring:

- Restructures the Compose UI components for better reusability that will be needed when implementing the next steps in the migration flow.
- Uses resource dimensions to calculate size of async images (user avatar & site icons).

> 🏁 **All these changes are behind a temporary feature flag.**
>
> 🔔 **Notice:** Once the feature flag is enabled the migration flow will always be presented. The Jetpack app build has to be uninstalled (or "Clear Data" should be done) in order to revert back to the normal functionality. Further development on the data layer will make sure the migration is not attempted again once it's successful. To be partially handled in #17406.

<details>
<summary>ℹ️ Remark about initializing Welcome Screen state (click to expand)</summary>

Triggering the right UI update after both data sets (account & sites) are received from the event bus is a bit tricky since these are two separate async events.  
I proceeded forward with a solution that initializes the Welcome Screen ui state if both account userName and sites list are not empty, though suggestions on how to improve this are more than welcome.

In my tests the site list was always loaded after account info, but I'm not sure this can be guaranteed, my knowledge on how the event bus works and how user data gets loaded is rather limited. Though if it could be reliable to assume the site list is always loaded after the account info that would simplify things.

To be noted that we'll probably change this entirely when the data layer work concludes, and the previous solution was also rather temporary.

---
</details>

To test:

0. Open the Jetpack app & login
1. Go to `Me` → `App Settings` → `Debug Settings`
2. Enable the `JetpackMigrationFlowFeatureConfig` & restart the app.
3. **Expect** to see the "Welcome to Jetpack!" screen
4. **Verify** that the UI matches the updated design
5. Tap the `Continue` button
6. **Verify** that the UI matches the design for the processing state:
   - The UI elements should be semi-transparent, except the continue button
   - The continue button should have a loading spinner
   - The site list should not be scrollable
   - Buttons should be disabled
   - **Note** As a temporary measure the processing state reverts after 5 seconds. This state will represent the data migration progress.

## Previews

<details open>
<summary>1️⃣ <b>Initial State</b></summary>

| Light | Dark |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/200000811-8e9f033f-53cc-436b-8205-2fd84ec276c5.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/200001327-7e3f25bd-6664-4953-b211-094d6e5237cd.png"> |
</details>

<details>
<summary>2️⃣ <b>In Progress/Processing State</b></summary>

| Light | Dark |
| - | - |
| <img width="314" src="https://user-images.githubusercontent.com/4588074/200003795-ac2140c7-e755-4719-a3ac-144699712f67.png"> | <img width="314" src="https://user-images.githubusercontent.com/4588074/200001836-5d9b68ab-6148-472d-a6a6-350db30cd065.png"> |
</details>

<details>
<summary>🎬 <b>Video Preview</b></summary>

https://user-images.githubusercontent.com/4588074/200006345-28db4e57-85d9-4116-9f22-06e9f6b8b66c.mp4

</details>

## Regression Notes
1. Potential unintended areas of impact
   None - the changes are behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

3. What automated tests I added (or what prevented me from doing so)
   Will add unit tests later when the integration with the data layer can be done.
   At the moment a big part of how the data is loaded is a temporary solution.

PR submission checklist:
